spring-cloud-stream/src/main/java/org/springframework/cloud/stream/config/BinderFactoryConfiguration.java

```diff
@@ -114,6 +117,13 @@ public BinderTypeRegistry binderTypeRegistry(ConfigurableApplicationContext conf
 		return new DefaultBinderTypeRegistry(binderTypes);
 	}
 
+	@Bean
+	@ConditionalOnEnabledHealthIndicator("binders")
+	@ConditionalOnMissingBean(name = "bindersHealthIndicator")
```

Comment:
No, this just follows the general pattern of other autoconfiguration beans. Also, endpoint usage is not conditional of enabling web endpoints.